### PR TITLE
Improve SearchBar (#404, language independent search)

### LINKED ISSUE #404: SearchBar: filterPredicate logic mistake
#### General information

* **0.5.12** (master, commit c0d69da11308d57) 

#### Description
In commit `5c47776 (Migrate SearchBar back)` filter code 

```
if (s.isBlank())
   return@Predicate true
else
   return@Predicate item.label.toLowerCase().contains(s.toLowerCase())
```
convert to
```
if (s.isEmpty())
  return true;
else if (item._label.toLowerCase().contains(s.toLowerCase()))
   return false;

```

and it's look like mistake.

### PR DESCRIPTION
**Pull request by pinkiesky**
1. Fixed #404 
2. Improved search/filtering in SearchBar: for search, a universal application name is now used, not just localized. See images below.

screenshot_20180620-125423
Before Search improvement. Oh ah! Another keymap required...  Only "Фонарик" app name available for us.

screenshot_20180620-130024
After Search improvement. We can use "flashlight" word, not only localized name!

### CODE REVIEW

File: app/src/main/java/com/benny/openlauncher/model/App.java [hunk 1]
```diff
@@ -51,4 +80,17 @@ public String getClassName() {
...
+     * App label for root locale.
+     * @see Locale#ROOT
+     */
+    @Nullable
+    public String getUniversalLabel() {
+        return _universalLabel;
+    }
+
+    public void setUniversalLabel(@Nullable String _universalLabel) {
+        this._universalLabel = _universalLabel;
```
Comment (gsantner) [inline app/src/main/java/com/benny/openlauncher/model/App.java]: _unive = unive;

File: app/src/main/java/com/benny/openlauncher/model/App.java [hunk 2]
```diff
@@ -16,6 +26,25 @@ public App(PackageManager pm, ResolveInfo info) {
...
+
+    private void updateUniversalLabel(PackageManager pm, ResolveInfo info) throws PackageManager.NameNotFoundException {
+        ApplicationInfo appInfo = info.activityInfo.applicationInfo;
+
+        Configuration config = new Configuration();
+        config.locale = Locale.ROOT;
+
+        Resources resources = pm.getResourcesForApplication(appInfo);
+        resources.updateConfiguration(config, null);
+
```
Comment (gsantner) [inline app/src/main/java/com/benny/openlauncher/model/App.java]: do we need the public setter at all? if only modification happens here, delete setter and set directly here

File: app/src/main/java/com/benny/openlauncher/viewutil/IconLabelItem.java [hunk 3]
```diff
@@ -69,6 +76,11 @@ public IconLabelItem(Context context, Drawable icon, String label, int forceSize
         _forceSize = forceSize;
     }
 
+    public IconLabelItem(Context context, Drawable icon, String label, @Nullable String searchInfo, int forceSize) {
+        this(context, icon, label, forceSize);
```
Comment (gsantner) [inline app/src/main/java/com/benny/openlauncher/viewutil/IconLabelItem.java]: no this.

File: app/src/main/java/com/benny/openlauncher/model/App.java (hunk 2 above)
Comment (pinkiesky) [reply app/src/main/java/com/benny/openlauncher/model/App.java]: You right, no need setter. Just my programming style. I find a style with a variable (constant, to be precise) more beautiful and comfortable in support.

File: app/src/main/java/com/benny/openlauncher/viewutil/IconLabelItem.java (hunk 3 above)
Comment (pinkiesky) [reply app/src/main/java/com/benny/openlauncher/viewutil/IconLabelItem.java]: What is the alternative? Repeat assignment? Can you please explain why `this` is a bad style?

In any case, of course, I will correct it. Is there a coding style for the project? In README, I found mention only of "Auto Reformat" in Android Studio.

Thank you

Comment (gsantner) [reply app/src/main/java/com/benny/openlauncher/viewutil/IconLabelItem.java]: `._searchInfo = searchInfo; `

File: app/src/main/java/com/benny/openlauncher/model/App.java (hunk 2 above)
Comment (pinkiesky) [reply app/src/main/java/com/benny/openlauncher/model/App.java]: `resources` is important in this code

Comment (gsantner) [reply app/src/main/java/com/benny/openlauncher/model/App.java]: didn't write anything about resources :D

File: app/src/main/java/com/benny/openlauncher/model/App.java [hunk 4]
```diff
@@ -90,7 +90,7 @@ public String getUniversalLabel() {
         return _universalLabel;
     }
 
-    public void setUniversalLabel(@Nullable String _universalLabel) {
-        this._universalLabel = _universalLabel;
+    public void setUniversalLabel(@Nullable String universalLabel) {
+        this._universalLabel = universalLabel;
```
Comment (gsantner) [inline app/src/main/java/com/benny/openlauncher/model/App.java]: _universalLabel = universalLabel

File: app/src/main/java/com/benny/openlauncher/viewutil/IconLabelItem.java [hunk 5]
```diff
@@ -77,8 +77,10 @@ public IconLabelItem(Context context, Drawable icon, String label, int forceSize
     }
 
     public IconLabelItem(Context context, Drawable icon, String label, @Nullable String searchInfo, int forceSize) {
-        this(context, icon, label, forceSize);
-        this._searchInfo = searchInfo;
+        _label = label;
```
Comment (gsantner) [inline app/src/main/java/com/benny/openlauncher/viewutil/IconLabelItem.java]: No! Call the constructor. Thats perfectly fine!
This was about the assignment!

**Comment by gsantner** [comment]
`_member = localvar`  not `this._member=`

File: app/src/main/java/com/benny/openlauncher/viewutil/IconLabelItem.java (hunk 5 above)
Comment (gsantner) [reply app/src/main/java/com/benny/openlauncher/viewutil/IconLabelItem.java]: Thanks!
Usually this one rule is in README somehow not in this project :D

**Pull request merged**